plugins/chat/app/services/chat_message_destroyer.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ChatMessageDestroyer
+  def destroy_in_batches(chat_messages_query, batch_size: 200)
+    chat_messages_query.in_batches(of: batch_size).each do |relation|
+      destroyed_ids = relation.destroy_all.pluck(:id)
+      reset_last_read_message_id(destroyed_ids)
+      delete_flags(destroyed_ids)
```

Comment:
I noticed we were not deleting flags for DMs before, is this intentional?

Reply:
It was missing. We introduced the ability to flag DMs recently, so it looks like I forgot add it then.